cantools/database/can/signal.py

```diff
@@ -247,12 +249,26 @@ def __init__(self,
         #: ``True``.
         self.is_signed: bool = is_signed
 
-        #: The initial value of the signal, or ``None`` if unavailable.
-        self.initial: Optional[int] = initial
+        # deprecated `initial` attribute
+        self._initial: Optional[Union[int, SignalValueType]] = raw_initial
```

Comment:
AFAIK, internal values can also be `float` or `bytes`. (not sure if the latter is currently implemented by anything, though...)

also -- because some file formats currently pass internal and others physical values here -- we need to keep this as a separate constructor argument, e.g., `legacy_initial`...